### PR TITLE
Assessment page redesign

### PR DESCRIPTION
**Pull request by nayak16**
Added Danger Zone and Material Design components

PR with Autolab2.0

screen shot 2016-12-18 at 4 33 27 pm

### CODE REVIEW

**Comment by nayak16** [comment]
@zyx-billy Can you look over this?

File: app/assets/javascripts/assessments_show.coffee [hunk 1]
```diff
@@ -8,3 +8,5 @@ $('#handin_show_assessment #fake-submit').click (e)->
 $("input[type='file']").change (e)->
 	$('#handin_show_assessment #new_submission').submit()
 	$("#handin_show_assessment input[type='file']").val("")  # clear for re-selecting the same file
+
+$('#confirmModal1').leanModal();
```
Comment (zyx-billy) [inline app/assets/javascripts/assessments_show.coffee]: @nayak16 did this leanModal work for you? I don't see anything pop up after clicking "release all grades". Are the script/styles required for this plugin included?

File: app/assets/javascripts/app-level.js [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
+	/* Adding CSRF token to every Ajax call by default */
+	jQuery.ajaxSetup({
+    	headers: {
+        	'X-CSRF-Token' : $('meta[name="csrf-token"]').attr('content')
+    	},
+	});
 
+	/* Activating Tooltips */
+	$(function () {
+  		$('[data-toggle="tooltip"]').tooltip()
```
Comment (zyx-billy) [inline app/assets/javascripts/app-level.js]: I know you didn't add this, but I just noticed this tooltip shows up as a blank square for me when hovering over attachments. This does seem to be bootstrap thing? Perhaps we should just replace/remove it.

File: app/assets/javascripts/assessments_show.coffee (hunk 1 above)
Comment (nayak16) [inline app/assets/javascripts/assessments_show.coffee]: Materialize modal didnt work. I'm using browser default.

File: app/assets/javascripts/app-level.js (hunk 2 above)
Comment (nayak16) [inline app/assets/javascripts/app-level.js]: Got rid of these

File: app/assets/javascripts/app-level.js [hunk 3]
```diff
@@ -19,9 +19,7 @@ $(document).ready(function() {
 	});
 
 	/* Activating Tooltips */
-	$(function () {
-  		$('[data-toggle="tooltip"]').tooltip()
-	})
+	$('.tooltipped').tooltip({delay: 50});
```
Comment (zyx-billy) [inline app/assets/javascripts/app-level.js]: The problem I was talking about was this (the blank black tooltip):
assessment_show_empty_tooltip
There already seems to be a tooltip so we could just get rid of this thing entirely.

File: app/assets/javascripts/assessments_show.coffee (hunk 1 above)
Comment (zyx-billy) [inline app/assets/javascripts/assessments_show.coffee]: Cool these work perfectly now

**Comment by nayak16** [comment]
@zyx-billy Finally fixed everything. Let me know if this looks good. Want to get this merged in ASAP

**Comment by zyx-billy** [comment]
@nayak16 Yeah, the blank tooltip is fixed (though there's now two tooltips on the same element, counting the default one given by 'title', but I guess that's not too big of an issue for now?) Let's get this merged then.

**Pull request merged**